### DISCUSSION TITLE
Post-processing operation：Result file of extracting fepx by grain

### DISCUSSION THREAD

**Discussion by Ming-is** [neperfepx/neper#506] [Q&A] [answered]
Fepx's simulation output files are output by element or node. I can't know which grain these elements and nodes correspond to. How can I obtain the stress or strain tensor of the specific grains?

And if I want to get the lattice strain corresponding to a specific fiber orientation, what should I do？

**Comment by rquey** [comment, marked as answer]
> Fepx's simulation output files are output by element or node. I can't know which grain these elements and nodes correspond to. How can I obtain the stress or strain tensor of the specific grains?

Just ask for it... If `stress` is defined for elements, you can get it for elsets (which represent grains) using `-reselset stress`.


> And if I want to get the lattice strain corresponding to a specific fiber orientation, what should I do？

You first have to define the fiber as an *entity*, as in `-entity "myfiber:fiber(1,0,0,1,0,0,15)"`.  This will define the fiber as the set of elements whose orientations are within 15° of the (100)||X orientation fiber. This also automatically defines new options, such as `-resmyfiber`. So, you can do, for example, `-resmyfiber stress` to get the average stress of `myfiber`.

**Reply by Ming-is** [reply]
OK, thanks, and do these instructions run in the S module?

**Reply by rquey** [reply]
Yes. All mesh keys can be used in -S.

**Reply by Ming-is** [reply]
For the second question, I used this instruction-------【 neper -S 4.sim -entity "myfiber:fiber(1,0,0,1,0,0,15)" -resmyfiber stress】，
but the program reported an error：

Info   : Running post-processing...
Info   :   - Loading inputs...
Info   :     > simulation.tess...
Info   :     > simulation.msh...
Info   :     > simulation.config...
Info   :   - Writing results...
Info   :     [o] Writing directory `4.sim/results/myfiber'...
Error  :     > Entity `myfiber' has no parent.

**Reply by rquey** [reply]
Update to version 4.4.2-29.

**Reply by Ming-is** [reply]
ok，thanks
